python/lsst/analysis/ap/apdb.py

```diff
@@ -228,7 +228,8 @@ def load_object(self, id, latest=True):
             The requested object.
         """
         table = self._tables["DiaObject"]
-        query = table.select().where(table.columns["validityStart"].isnot(None))
+        if latest:
+            query = table.select().where(table.columns["validityEnd"] == None)  # noqa: E711
```

Comment:
Looking at this again, I just realized that for `load_object`, if `latest=False`, this will return whichever object is `[0]` in the sql query result (see the `return` statement). The intent of `load_object` is to give you the single row for that object, and `latest=False` changes that. I guess my test data only includes one diaObject here, so it can't check for whether the correct "version" of it was returned, nor whether was multiple or just one .

I'm not sure what the best approach is here: having the return sometimes be a `pd.Series` and sometimes be `pd.DataFrame` is potentially confusing (even if the user is the one causing the change by setting a kwarg), but there doesn't see much point to adding a new method. What is the use case for getting all of the non-valid versions of a DiaObject here?

Reply:
Good point. I modified `load_object` so it simply loads the "latest" version of the diaObject in question (i.e., the one with validityEnd equal to None). This should have consistent behavior --- return one record --- and work for the majority of use cases.